app/controllers/user_badges_controller.rb

```diff
@@ -116,4 +119,8 @@ def can_assign_badge_to_user?(user)
   def ensure_badges_enabled
     raise Discourse::NotFound unless SiteSetting.enable_badges?
   end
+
+  def is_badge_reason_valid?(reason)
+    return Discourse.route_for(reason)
```

Comment:
In Ruby, you don't need the `return` as it's always returning the value of the last statement in a method/function.

You need to check that the route is one that points to a post or a topic. Not all routes in Discourse are valid for this badge ;)